src/types.js

```diff
@@ -97,11 +98,13 @@
  * @property {ActualCard[]} [linked]
  * @property {boolean} [certain]
  * @property {boolean} [ambiguous]
+ * @property {Connection[]} [self_connection]
```

Comment:
Does `self_connection` on Connection/FocusPossibility need to store the entire Connection array? It seems like only the existence of such connections is all that's needed. If so, these could be changed to booleans.

Reply:
I was originally thinking we'd need to track the card so that if we later determined we couldn't have the card we could update the condition, but i guess this would happen even with a boolean through the rewind and replay.